src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -151,50 +154,31 @@ ConcurrencyWorker::PrepAndSendInferRequests()
   // Non-sequence model is 'num_reqs' * 1 ctx
   // Sequence model is 1 request of 1 sequence * 'active_ctx_cnt' ctxs_
   while (total_ongoing_requests_ < (int)thread_config_->concurrency_ &&
-         early_exit == false) {
-    PrepAndSendInferRequest();
+         early_exit == false && execute_) {
```

Comment:
Why execute was added:
For async, we will tally up total_ongoing_requests, and exit this while loop as appropriate
For sync, we no longer tally up total (because sync can only ever have one ongoing at a time). We used to still count up the number of sync requests we sent to exit the loop, but really we just need to exit when we are done executing

Reply:
So execute is just being added here to prevent threads from running this portion of the code while we are between concurrencies/request rates?